### PR TITLE
CLOUDP-159055: Support copySettings field for BackupSchedule

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: pkg/controller/validate/validate.go [hunk 1]
```diff
@@ -60,6 +60,49 @@ func DatabaseUser(_ *mdbv1.AtlasDatabaseUser) error {
 	return nil
 }
 
+func BackSchedule(bSchedule *mdbv1.AtlasBackupSchedule, deployment *mdbv1.AtlasDeployment) error {
```
Comment (igor-karpukhin) [inline pkg/controller/validate/validate.go:63]:
```suggestion
func BackupSchedule(bSchedule *mdbv1.AtlasBackupSchedule, deployment *mdbv1.AtlasDeployment) error {
```

File: pkg/controller/atlasdeployment/backup.go [hunk 2]
```diff
@@ -88,7 +89,11 @@ func (r *AtlasDeploymentReconciler) ensureBackupSchedule(
 		return nil, errors.New(errText)
 	}
 
-	bSchedule.UpdateStatus([]status.Condition{}, status.AtlasBackupScheduleSetDeploymentID(clusterName))
+	if err = validate.BackSchedule(bSchedule, deployment); err != nil {
```
Comment (igor-karpukhin) [inline pkg/controller/atlasdeployment/backup.go:92]:
```suggestion
	if err = validate.BackupSchedule(bSchedule, deployment); err != nil {
```

**Review by igor-karpukhin — CHANGES_REQUESTED** [review]
Looks good in general, please update integration test as well (deployment_test.go)

**Review by igor-karpukhin — APPROVED** [review]
Looks good 👍

**Review by fabritsius — APPROVED** [review]
Looks Great 👍

**Pull request merged**